java/src/main/java/sort/BubbleSort.java

```diff
@@ -0,0 +1,52 @@
+package sort;
+
+import java.util.List;
+import javafx.util.Pair;
+import sort.Options.ORDER;
+
+public class BubbleSort {
+  static <T> Pair<T, T> getSwapped(Pair<T, T> pair) {
+    return new Pair<T, T>(pair.getValue(), pair.getKey());
+  }
+
+  /*
+  Classic bubble sort works by
+  - Iterating over each element
+  - For each other element, swap if out of order
+  - Optimization: if each other element is already ordered, stop iterating
```

Comment:
That line got me a bit confused :sweat_smile:

Reply:
How about: "if no swaps are needed, we consider the list to be ordered and thus stop iterating" ?

Reply:
Sure :)